### PR TITLE
add touchable property to actor

### PR DESCRIPTION
**Pull request by dasg34**
1. Rename toucalbe classes
 - NearInteractionTouchable -> TouchablePlane
 - NearInteractionTouchableSurface -> ITouchableSurface
 - BaseNearInteractionTouchable -> ITouchableBase
2. Modify PokeTool logic to fix some issues.
3. Add touchable property to actor

Sample code of touchable actor:
```ts
MRE.Actor.Create(this.context, {
    actor: {
        collider: {
            geometry: {
                shape: MRE.ColliderType.Box,
                size: {x: 0.1, y: 0.1, z : 0.1}
            }
        },
        touchable: {
            type: MRE.TouchableType.Surface,
            direction: MRE.TouchableDirection.Right
        }
    }
});
```

`TouchableType.Volume` is not supported yet.
`TouchableType.Surface` works with box colldier. `touchable.direction`
is local forward directio of box collider. it needs to face camera.
`touchable.bounds` is the touchable bounds of the surface.
When actor is touchable, `ButtonBehavior` will function with poketool
(index finger).

### CODE REVIEW

**Comment by dasg34** [comment]
Touchable Actor Preview :)
TouchableActor

Sample Code 
```ts
const actor = MRE.Actor.Create(this.context, {
	actor: {
		name: "touchble actor",
		appearance: {
			meshId: this.assets.createBoxMesh("cube", 0.1, 0.1, 0.1).id,
		},
		collider: {
			geometry: {
				shape: MRE.ColliderType.Box,
				size: {x: 0.1, y: 0.1, z : 0.1}
			}
		},
		touchable: {
			type: MRE.TouchableType.Surface,
			bounds: {x: 0.1, y:0.1},
			direction: MRE.TouchableDirection.Forward
		}
	}
})
const behavior = actor.setBehavior(MRE.ButtonBehavior);
behavior.onButton("pressed", () => {
	console.log("pressed");
});
behavior.onButton("released", () => {
	console.log("unpressed");
});
```

**Comment by woohyun** [comment]
I'm not clear with "touchable.bounds". Could it be handled by changing the size of collider instead, nope ?

**Review by woohyun — APPROVED** [review]
Except some trivial things, I think everything looks good :-)

**Comment by dasg34** [comment]
> I'm not clear with "touchable.bounds". Could it be handled by changing the size of collider instead, nope ?

That's true. "touchable.bounds" has been removed with new commit.
Also, server api can be simplified.
Thank you for the reviewing.

File: MREGodotRuntime/Scripts/Tools/PokeTool.cs [hunk 1]
```diff
@@ -13,25 +15,25 @@ public class PokeTool : Tool
...
+		public ITouchableBase ClosestProximityTouchable { get; private set; }
 		public float TouchableDistance { get; } = 0.2f;
 		public Spatial CurrentTouchableObjectDown { get; private set; }
 		public Vector3 PreviousPosition { get; private set; } = Vector3.Zero;
-		public Vector3 IntersectionPosition { get; private set; }
 
 		private Spatial CurrentPointerTarget;
 		private Vector3 RayStartPoint;
 		private Vector3 RayEndPoint;
+		private float previousClosestDistance;
```
[the remark this answers is not in the document]
Comment (dasg34) [reply MREGodotRuntime/Scripts/Tools/PokeTool.cs]: ` = 0.0f;` is added.

**Review by woohyun — APPROVED** [review]
Thanks for getting my feedback. All good for me now :)

**Pull request merged**